### DISCUSSION TITLE
Video playback stuttering while playing beatmap

### DISCUSSION THREAD

**Discussion by KeyboardCombination** [ppy/osu#23966] [Q&A]
### Type

Game behaviour

### Bug description

I noticed this stuttering while playing beatmaps with video playback enabled, the stuttering happens occasionally across the entire song seemingly at random. Everything else runs just fine.

I am playing the game on a 2020 M1 MacBook Air.

### Screenshots or videos

https://github.com/ppy/osu/assets/44244130/c25a1447-54cc-4801-8d6b-bdd61e73285f

### Version

2023.605.0

### Logs

[database.log](https://github.com/ppy/osu/files/11709341/database.log)
[input.log](https://github.com/ppy/osu/files/11709342/input.log)
[network.log](https://github.com/ppy/osu/files/11709343/network.log)
[performance.log](https://github.com/ppy/osu/files/11709344/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11709346/runtime.log)

**Comment by peppy** [comment]
Quite weird. This used to happen in stable too.

We should at very least make sure the video playback doesn't return to a previous frame if it's running behind.

A few questions:

- Did you try disabling video hardware acceleration in settings to see if it helps?
- Are you running in "low power" mode (macOS setting)
- Does this happen for all beatmaps or just the one you show in the video? Can you link the beatmaps it happens in if it's not all beatmaps with video?

**Comment by KeyboardCombination** [comment]
- Turning off hardware acceleration doesn't help at all
- I don't have "low power" mode on macOS whatsoever, I have it set to never
- I've tried a few other beatmaps, same issue

**Comment by 2minwia** [comment]
I'd like to add that this happens to me too, if a larger sample size can help out in any way :)
